sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -240,6 +240,7 @@ class BucketedReadSuite extends QueryTest with SQLTestUtils with TestHiveSinglet
       joinCondition: (DataFrame, DataFrame) => Column,
       shuffleLeft: Boolean,
       shuffleRight: Boolean,
+      numPartitions: Int = 10,
```

Comment:
Since other vars are for left and right side, wondering if same could to be done for `numPartitions`. ie. `numPartitionsLeft` and `numPartitionsRight`. 

That could make the new test you added more interesting : 

```
 numPartitionsLeft = 1,
 numPartitionsRight = 10,
 sortLeft = false,
 sortRight = true
```